### PR TITLE
Fixes #12479 - Fixed issue where time lookups on a symlink (GetLastWrite, Creation, …

### LINKED ISSUE #12479: File.GetLastWriteTime behaves differently for symlinks, as compared to netcore on macOS
## Steps to Reproduce

1. Create a symlink to a file (ensure that the timestamp differs from the symlink)
```
Jan 17 15:23 link@ -> symlink.csproj
Jan 17 15:22 symlink.csproj ## Target file
```

2. Run the test code with two files.

### Current behavior:
Mono returns the same timestamp, that for the target file.
```
$ mono symlink.exe link
File.GetLastWriteTime for link : 1/17/2019 3:22:08 PM

$ mono symlink.exe link
File.GetLastWriteTime for symlink.csproj : 1/17/2019 3:22:08 PM
```

dotnet returns the timestamp for the symlink itself when passed that.
```
$ dotnet run -- link
File.GetLastWriteTime for link : 1/17/19 3:23:36 PM

 $ dotnet run -- symlink.csproj
File.GetLastWriteTime for symlink.csproj : 1/17/19 3:22:08 PM
```

### Expected behavior:
The behavior should match between the two.

```
using System;
using System.IO;

class Program
{
        static void Main(string[] args) =>  Console.WriteLine ($"File.GetLastWriteTime for {args[0]} : {File.GetLastWriteTime(args[0])}");
}
```

/cc @marek-safar

### PR DESCRIPTION
**Pull request by steveisok**
When given a path to a symlink `GetLastWriteTime` returned the time for the actual file.  This PR resolves that.

Fixes https://github.com/mono/mono/issues/12479

### CODE REVIEW

File: mono/metadata/w32file-unix.c [hunk 1]
```diff
@@ -3578,26 +3579,33 @@ mono_w32file_get_attributes_ex (const gunichar2 *name, MonoIOStat *stat)
 	stat->attributes = _wapi_stat_to_file_attributes (utf8_name, &buf, &linkbuf);
 	stat->length = (stat->attributes & FILE_ATTRIBUTE_DIRECTORY) ? 0 : buf.st_size;
 
+	if (&linkbuf == NULL) {
+		pbuf = &buf;
```
Comment (lewurm) [inline mono/metadata/w32file-unix.c]: `&linkbuf` will _never_ be `NULL` as it's allocated on the stack so it always has an address. In the end, your change boils down to `pbuf = &linkbuf` all the time. And this is fine, as `lstat` behaves the same as `stat` if the provided file isn't a symbolic link.

**Comment by lewurm** [comment]
I pushed a cleanup commit to address the (cosmetic) issue I mentioned. Thanks for your contribution!

**Comment by lewurm** [comment]
@monojenkins squash

**Comment by marek-safar** [comment]
Thank you for your contribution.

**Pull request merged**